prime-dotnet-webapi/Services/EnrolleeAgreementService.cs

```diff
@@ -171,5 +178,30 @@ private async Task RenderHtml(params Agreement[] agreements)
                 }
             }
         }
+
+        public async Task<bool> IsAgreementTypeIdenticalAsync(int enrolleeId)
+        {
+            var currentAgreement = await GetCurrentAgreementAsync(enrolleeId);
+
+            // Get only the information needed to determine expected agreement type
+            var enrollee = await _context.Enrollees
+                .AsNoTracking()
+                .Include(e => e.Certifications)
+                    .ThenInclude(c => c.License)
+                .Include(e => e.EnrolleeCareSettings)
+                .SingleOrDefaultAsync(e => e.Id == enrolleeId);
+            foreach (var cert in enrollee.Certifications)
+            {
+                cert.License.LicenseDetails = await _context.Set<LicenseDetail>()
+                    .OrderByDescending(ld => ld.EffectiveDate)
+                    .Where(ld => ld.LicenseCode == cert.License.Code)
+                    .Where(ld => ld.EffectiveDate <= DateTime.UtcNow)
+                    .ToListAsync();
```

Comment:
I don't see why this `foreach` is necessary, why cant you just `.Include` the license details?